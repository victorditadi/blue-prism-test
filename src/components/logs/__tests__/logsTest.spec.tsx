import React from 'react';
import { render } from '@testing-library/react';

import Theme from 'css/theme';

import { logsMock } from './logsMock';
import { Logs } from '..';
import { QueryClient, QueryClientProvider } from 'react-query';

describe('Component: Logs', () => {
  const renderComponent = ({
    isError = false,
    isLoading = false,
  }: {
    isError?: boolean;
    isLoading?: boolean;
  }) => {
    const queryClient = new QueryClient();

    return render(
      <QueryClientProvider client={queryClient}>
        <Theme>
          <Logs isErrorLogs={isError} isLoadingLogs={isLoading} logs={logsMock} />
        </Theme>
      </QueryClientProvider>
    );
  };

  it('should render Logs component', () => {
    const { asFragment } = renderComponent({});

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render skeleton if is loading', () => {
    const { queryByTestId } = renderComponent({ isLoading: true });
    expect(queryByTestId('logs-skeleton')).toBeInTheDocument();
  });

  it('should not render skeleton if is not loading', () => {
    const { queryByTestId } = renderComponent({});
    expect(queryByTestId('logs-skeleton')).not.toBeInTheDocument();
  });

  it('should render error if have error', () => {
    const { queryByTestId } = renderComponent({ isError: true });
    expect(queryByTestId('logs-error')).toBeInTheDocument();
  });
});
