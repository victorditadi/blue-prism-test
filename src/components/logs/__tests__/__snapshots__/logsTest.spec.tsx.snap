// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Logs should render Logs component 1`] = `
<DocumentFragment>
  .c4 {
  width: calc(100% * (1 / 4) - 20px);
  height: calc(100%);
  border: 1px solid #d0e5ed;
  margin-bottom: 1vw;
  padding: 1vw;
  background-color: #e8f5f9;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-flow: wrap;
  -ms-flex-flow: wrap;
  flex-flow: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-align: center;
}

.c6 {
  -webkit-flex: 50%;
  -ms-flex: 50%;
  flex: 50%;
  font-family: Robotosans-serif;
  font-size: 1em;
  margin-top: 8px;
  position: relative;
}

.c7 {
  font-family: Robotosans-serif;
  font-weight: bold;
  font-size: 1em;
}

.c10 {
  font-size: 1em;
  font-family: Robotosans-serif;
}

.c12 {
  visibility: hidden;
  display: none;
}

.c8 .c9:hover ~ .c11 {
  visibility: visible;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: absolute;
  font-family: Robotosans-serif;
  font-size: 1em;
  background-color: #e8f5f9;
  width: 100%;
  padding: 10px 10px;
  border-radius: 4px;
  border: 1px solid #d0e5ed;
  z-index: 9999;
}

.c0 {
  height: 70vh;
  position: relative;
  margin: 0 5vw 0 5vw;
  width: 100%;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  max-height: 100%;
  overflow: auto;
  overflow-y: scroll;
}

.c3:after {
  content: '';
  width: 29.5vw;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 1vw;
}

.c2 {
  font-family: Robotosans-serif;
  font-weight: bold;
  font-size: 2em;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        Logs
      </span>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <span
            class="c6"
            data-testid="log-id-container"
          >
            <span
              class="c7"
              data-testid="log-id-type"
            >
              Id:
            </span>
            <div
              class="c8"
              data-testid="log-id-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                88578155
              </span>
            </div>
          </span>
          <span
            class="c6"
            data-testid="log-startTime-container"
          >
            <span
              class="c7"
              data-testid="log-startTime-type"
            >
              Start Time:
            </span>
            <div
              class="c8"
              data-testid="log-startTime-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                10/15/2021
              </span>
            </div>
          </span>
          <span
            class="c6"
            data-testid="log-endTime-container"
          >
            <span
              class="c7"
              data-testid="log-endTime-type"
            >
              End Time:
            </span>
            <div
              class="c8"
              data-testid="log-endTime-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                6/9/2021
              </span>
            </div>
          </span>
          <span
            class="c6"
            data-testid="log-status-container"
          >
            <span
              class="c7"
              data-testid="log-status-type"
            >
              Status:
            </span>
            <div
              class="c8"
              data-testid="log-status-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                Running
              </span>
            </div>
          </span>
          <span
            class="c6"
            data-testid="log-serverName-container"
          >
            <span
              class="c7"
              data-testid="log-serverName-type"
            >
              Server Name:
            </span>
            <div
              class="c8"
              data-testid="log-serverName-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                cillum
              </span>
            </div>
          </span>
          <span
            class="c6"
            data-testid="log-scheduleId-container"
          >
            <span
              class="c7"
              data-testid="log-scheduleId-type"
            >
              Schedule Id:
            </span>
            <div
              class="c8"
              data-testid="log-scheduleId-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                81112054
              </span>
            </div>
          </span>
        </div>
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <span
            class="c6"
            data-testid="log-id-container"
          >
            <span
              class="c7"
              data-testid="log-id-type"
            >
              Id:
            </span>
            <div
              class="c8"
              data-testid="log-id-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                36629794
              </span>
            </div>
          </span>
          <span
            class="c6"
            data-testid="log-startTime-container"
          >
            <span
              class="c7"
              data-testid="log-startTime-type"
            >
              Start Time:
            </span>
            <div
              class="c8"
              data-testid="log-startTime-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                8/5/2021
              </span>
            </div>
          </span>
          <span
            class="c6"
            data-testid="log-endTime-container"
          >
            <span
              class="c7"
              data-testid="log-endTime-type"
            >
              End Time:
            </span>
            <div
              class="c8"
              data-testid="log-endTime-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                4/6/2021
              </span>
            </div>
          </span>
          <span
            class="c6"
            data-testid="log-status-container"
          >
            <span
              class="c7"
              data-testid="log-status-type"
            >
              Status:
            </span>
            <div
              class="c8"
              data-testid="log-status-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                Exception
              </span>
            </div>
          </span>
          <span
            class="c6"
            data-testid="log-serverName-container"
          >
            <span
              class="c7"
              data-testid="log-serverName-type"
            >
              Server Name:
            </span>
            <div
              class="c8"
              data-testid="log-serverName-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                deserunt p...
              </span>
              <div
                class="c11 c12"
                data-testid="tooltip"
              >
                deserunt pariatur
              </div>
            </div>
          </span>
          <span
            class="c6"
            data-testid="log-scheduleId-container"
          >
            <span
              class="c7"
              data-testid="log-scheduleId-type"
            >
              Schedule Id:
            </span>
            <div
              class="c8"
              data-testid="log-scheduleId-value"
            >
              <span
                class="c9 c10"
                data-testid="short-text"
              >
                52888597
              </span>
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
